### DISCUSSION TITLE
Is it possible to set custom CPU and memory resources for in built filters/sources/sinks

### DISCUSSION THREAD

**Discussion by louisnow** [numaproj/numaflow#729] [General]
Is it possible to set custom CPU and memory resources for in built filters/sources/sinks?

I understand it's possible for custom UDF where we specify the contain template but it doesn't look like I can set custom k8s resources for the in-built filters, sources and sinks. This would allow us to highly fine tune each pipeline.

Is there a way to maybe do this at an individual pipeline level if not at a vertex level?

**Comment by louisnow** [comment]
Oops, I missed it because I was looking at the CRD defintion but it's present in the docs. 

I believe this is at the pipeline level https://numaflow.numaproj.io/user-guide/reference/configuration/pipeline-customization/

While this is at the vertex level https://numaflow.numaproj.io/user-guide/reference/configuration/container-resources/

**Reply by whynowy** [reply]
👍

**Comment by whynowy** [comment]
@louisnow - do you mind putting your company name in https://github.com/numaproj/numaflow/blob/main/USERS.md?
